dnf/base.py

```diff
@@ -2427,23 +2427,6 @@ def _prov_key_data(msg):
             errmsg = ucd(errmsg)
             raise dnf.exceptions.Error(_prov_key_data(errmsg))
 
-    def package_import_key(self, pkg, askcb=None, fullaskcb=None):
-        # :api
-        """Retrieve a key for a package. If needed, use the given
-        callback to prompt whether the key should be imported.
-
-        :param pkg: the package object to retrieve the key of
-        :param askcb: Callback function to use to ask permission to
-           import a key.  The arguments *askcb* should take are the
-           package object, the userid of the key, and the keyid
-        :param fullaskcb: Callback function to use to ask permission to
-           import a key.  This differs from *askcb* in that it gets
-           passed a dictionary so that we can expand the values passed.
-        :raises: :class:`dnf.exceptions.Error` if there are errors
-           retrieving the keys
-        """
-        _get_key_for_package(pkg, ackcb, fullaskcb)
```

Comment:
Sorry, this is my fault. The code indeed does not work and was added as part of fix for bug https://bugzilla.redhat.com/show_bug.cgi?id=1793298. There is also quite old bug https://bugzilla.redhat.com/show_bug.cgi?id=1339617 that requires API for GPG keys handling. Shouldn't we rather fix the code then remove it? It is also one of functions that (due to lack of proper API) the dnfdaemon keeps copy of (see  https://github.com/manatools/dnfdaemon/blob/87f92586ef1545b9d9118dcb7b093193f3230a89/python/dnfdaemon/server/backend.py#L220)

Reply:
Fixed; the method is no longer removed.